src/pke/lib/scheme/bgvrns/bgvrns-parametergeneration.cpp

```diff
@@ -445,7 +445,9 @@ bool ParameterGenerationBGVRNS::ParamsGenBGVRNS(std::shared_ptr<CryptoParameters
         qBound += cryptoParamsBGVRNS->EstimateMultipartyFloodingLogQ();
 
     // we add an extra bit to account for the special logic of selecting the RNS moduli in BGV
-    qBound++;
+    // ignore the case when there is only one max size modulus
+    if (qBound != static_cast<double>(auxBits))
```

Comment:
qBound and auxBits are of the same type - uint32_t. The cast to double is not needed.